go/libkb/log_profile.go

```diff
@@ -0,0 +1,126 @@
+package libkb
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+// LogProfileContext for LogProfile
+type LogProfileContext struct {
+	Contextified
+	Path string
+}
+
+func maxDuration(durations []time.Duration) time.Duration {
+	max := time.Duration(0)
+	for _, d := range durations {
+		if d > max {
+			max = d
+		}
+	}
+	return max
+}
+
+func minDuration(durations []time.Duration) time.Duration {
+	if len(durations) == 0 {
+		return 0
+	}
+	min := durations[0]
+	for _, d := range durations {
+		if d < min {
+			min = d
+		}
+	}
+	return min
+}
+
+func avgDuration(durations []time.Duration) time.Duration {
+	if len(durations) == 0 {
+		return 0
+	}
+	var total int64
+	for _, d := range durations {
+		total += d.Nanoseconds()
+	}
+	return time.Duration(total / int64(len(durations)))
+}
+
+func format(fn string, durations []time.Duration) string {
```

Comment:
It would be good to make all these helpers by receivers of LogProfileContext, just for namespacing. Especially this once since as it stands it's perched at `libkb.format`

Reply:
fixed